### PR TITLE
Implement #82

### PR DESCRIPTION
**Pull request by BinAccel**
Branch `rec` has errors, using this instead.

@Xyene @quantum5 plz delete `rec`. Ty

### CODE REVIEW

File: dmoj/judgeenv.py [hunk 1]
```diff
@@ -98,12 +98,31 @@ def load_env(cli=False, testsuite=False):  # pragma: no cover
 
         dirs = env.problem_storage_root
         if dirs is not None:
+            get_path = lambda x, y: unicodify(os.path.normpath(os.path.join(x, y)))
             if isinstance(dirs, ConfigNode):
-                problem_dirs = tuple(unicodify(os.path.normpath(os.path.join(_root, dir))) for dir in dirs)
+                import re
+                rerec = re.compile(r'^(\d+)\s*:\s*(.*)$')
+                # Allow for recursion
+                def rec(dir, dep):
```
Comment (quantum5) [inline dmoj/judgeenv.py]: Stop reinventing the wheel...

File: dmoj/judgeenv.py [hunk 2]
```diff
@@ -98,12 +98,31 @@ def load_env(cli=False, testsuite=False):  # pragma: no cover
...
+                    if not dep: return [dir]
+                    ret = []
+                    for d in os.listdir(dir):
+                        nxt = os.path.join(dir, d)
+                        if os.path.isdir(nxt):
+                            ret += rec(nxt, dep - 1)
+                    return ret
+
+                problem_dirs = []
```
Comment (quantum5) [inline dmoj/judgeenv.py]: Literally not how I said that issue should go. Arbitrary recursion is stupid since new directories won't be added.

Comment (BinAccel) [inline dmoj/judgeenv.py]: Recursion is controlled by depth, and it does pick up on new directories.

File: dmoj/judgeenv.py [hunk 3]
```diff
@@ -98,12 +98,29 @@ def load_env(cli=False, testsuite=False):  # pragma: no cover
 
         dirs = env.problem_storage_root
         if dirs is not None:
+            get_path = lambda x, y: unicodify(os.path.normpath(os.path.join(x, y)))
             if isinstance(dirs, ConfigNode):
-                problem_dirs = tuple(unicodify(os.path.normpath(os.path.join(_root, dir))) for dir in dirs)
+
+                def find_directories_by_depth(dir, depth):
+                    if not depth: return [dir] if os.path.isdir(dir) else []
```
Comment (quantum5) [inline dmoj/judgeenv.py]: Now just this awkward line.

**Comment by Xyene** [comment]
yolo

**Pull request merged**